src/main/kotlin/cacophony/automata/DFAJoin.kt

```diff
@@ -0,0 +1,83 @@
+package cacophony.automata
+
+import cacophony.automata.minimalization.via
+
+// Joins given list of the automata into a single DFA with accepting states marked
+// with a Result corresponding to the source DFAs.
+// Throws if any state would be marked by many results, i.e. when any two DFAs accept the same word.
+fun <State, Atom, Result> joinAutomata(
```

Comment:
in #59 we kind of embraced the convention of adding suffix `Type` to type arguments, so I'd stick to that, so
`StateType`, `AtomType` and `ResultType`